andela/tesla-ah
# #166705699 - Get an error for the unavailable URL

#### What does this PR do?
In this PR, I added a middleware that will track if the URL provided is available or reachable
#### Description of Task to be completed?
having the following endpoint, we return an error message to the user:
`http://localhost/fdfasfasfafasfaasffaf`
`http://localhost/api/fdfasfasfafasfaasffaf`
#### How should this be manually tested?
- clone the repository
- go to your terminal and then write the following command
  `cd tesla-ah`
- after that try to install all dependencies that are required for this project, simply by running the following command:
   `yarn install`  or `yarn`
- Then running the server, by running the below commands:
   1. `sequelize db:migrate` for this command make sure that the `sequelize-cli` package is installed globally
   2. `yarn dev`
   3.  After running the server, Open your postman, and then try to send a request with a bad URL
        you will get an error message
- for running the tests, write the below command in your terminal
     `yarn test`
 
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#166705699 ](https://www.pivotaltracker.com/n/projects/2350798/stories/166705699)
#### Screenshots (if appropriate)
N/A
#### Questions:








****
**Before Code - Audace Uhiriwe**
https://docs.google.com/document/d/1mEyswr6qQIkYr93ekXAXCK39t8toLdhu8NQ94n2ahXw/edit?usp=sharing
